File: app/src/main/java/com/aphoh/muser/base/TransformerContainer.java
package com.aphoh.muser.base;

import rx.Observable;
import rx.Observable.Transformer;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

/**
 * Created by Will on 11/14/15.
 */
public class TransformerContainer {
  public <T> Transformer<T, T> get() {
    return new Transformer<T, T>() {
      @Override public Observable<T> call(Observable<T> tObservable) {
        return tObservable.subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread());
      }
    };
  }
}

File: app/src/main/kotlin/com/aphoh/muser/module/TransformerContainer.kt
package com.aphoh.muser.module

import rx.Observable
import rx.Observable.Transformer
import rx.android.schedulers.AndroidSchedulers
import rx.schedulers.Schedulers

/**
 * Created by Will on 11/14/15.
 */
class TransformerContainer {
    fun <T> get(): Transformer<T, T> {
        return Transformer { tObservable -> tObservable.subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread()) }
    }
}
